src/JJMasterData.Web/Areas/MasterData/Controllers/OptionsController.cs

```diff
@@ -18,32 +17,21 @@ public OptionsController(OptionsService service)
     
     public async Task<IActionResult> Index(bool isFullscreen=false)
     {
-        var settings = new OptionsViewModel
-        {
-            ConnectionString = new ConnectionString(Service.Options.GetConnectionString()),
-            Options = Service.Options,
-            ConnectionProvider = DataAccessProvider.GetDataAccessProviderTypeFromString(Service.Options.GetConnectionProvider()),
-            FilePath = Service.JJMasterDataOptionsWriter.FilePath,
-            IsFullscreen = isFullscreen,
-        };
-
-        if (isFullscreen)
-        {
-            settings.ConnectionString.ConnectionResult = 
-                await GetConnectionResultAsync(settings.ConnectionString.ToString());
-        }
+        var viewModel = await Service.GetViewModel(isFullscreen);
 
-        return View(settings);
+        return View(viewModel);
     }
     
+
     public async Task<IActionResult> Save(OptionsViewModel model)
     {
         if (ModelState.IsValid)
```

Comment:
this conditional could be at line 32, because Service.SaveOptions(model) can change the ModelState